sr_object_symmetry/src/main.cpp

```diff
@@ -0,0 +1,318 @@
+  ros::param::get("~min_inlier_normal_angle", reflDetParams.min_inlier_normal_angle);
+  reflDetParams.min_inlier_normal_angle = pcl::deg2rad(reflDetParams.min_inlier_normal_angle);
+  ros::param::get("~max_inlier_normal_angle", reflDetParams.max_inlier_normal_angle);
+  reflDetParams.max_inlier_normal_angle = pcl::deg2rad(reflDetParams.max_inlier_normal_angle);
+  ros::param::get("~min_cloud_inlier_score", reflDetParams.min_cloud_inlier_score);
+  ros::param::get("~min_corresp_inlier_score", reflDetParams.min_corresp_inlier_score);
+  ros::param::get("~symmetry_min_angle_diff", reflDetParams.symmetry_min_angle_diff);
+  reflDetParams.symmetry_min_angle_diff = pcl::deg2rad(reflDetParams.symmetry_min_angle_diff);
+  ros::param::get("~symmetry_min_distance_diff", reflDetParams.symmetry_min_distance_diff);
+  ros::param::get("~max_reference_point_distance", reflDetParams.max_reference_point_distance);
+  // Generic patameters
+  ros::param::get("~visualization", visEnable);
+  ros::param::get("~overwrite", overWrite);
+  ros::param::get("~yaml_path", yamlPath);
+  ros::param::get("~objects_path", objectsPath);
+  ros::param::get("~object_sampling", objectSampling);
+  ROS_INFO("Parameters loaded");
+  // Check if objects_path is a single file or a folder
+  struct stat s;
+  struct dirent *dirp;
+  if (stat(objectsPath.c_str(), &s) == 0)
+  {
+    if (s.st_mode & S_IFDIR)
+    {
+      ROS_INFO("Folder scanning mode");
+      dir = opendir(objectsPath.c_str());
+      while ((dirp = readdir(dir)) != NULL)
+      {
+        std::string fname = dirp->d_name;
+        if ((fname.find(".stl") != std::string::npos) || (fname.find(".ply") != std::string::npos)
+                                                      || (fname.find(".obj") != std::string::npos))
+          files.push_back(objectsPath + "/" + fname);
+      }
+      closedir(dir);
+    }
+    else if (s.st_mode & S_IFREG)
+    {
+      files.push_back(objectsPath);
+      ROS_INFO("Single file defined");
+    }
+  }
+  else
+  {
+    ROS_INFO_STREAM("Error checking input path: " << objectsPath.c_str());
+    return 0;
+  }
+  for (size_t fileIndex = 0; fileIndex < files.size(); fileIndex++)
+  {
+    // Create new Symmetries object
+    SymmetryDetection symmetries_T;
+    reflDetParams.rot_symmetries.clear();
+    yamlFile = yamlPath + "/" + utl::getBasenameNoExtension(files[fileIndex].c_str()) + ".yaml";
+    // First check if file exists, if not then extract symmetries
+    if ((utl::isFile(yamlFile) == false) || (overWrite == true))
+    {
+      ROS_INFO_STREAM("Loading object: " << files[fileIndex].c_str());
+      symmetries_T.loadFile(files[fileIndex].c_str(), objectSampling);
+      // Detect rotational symmetries
+      symmetries_T.rotationalDetection<PointT>(rotDetParams);
+      if (symmetries_T.getRotational().size() == 0)
+      {
+        ROS_WARN("Could not find rotational symmetries");
```

Comment:
That shouldn't be a warning, just INFO, as it is completely normal.

Reply:
donw